### PR TITLE
fix: add proxy support

### PR DESCRIPTION
**Pull request by timja-kainos**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Needle doesn't support automatic proxy like request does, snyk moved to
needle on v1.41.0

This change will look for the standard unix environment variables for
proxying.

Note, I couldn't see any support for no_proxy in needle, so the docs
here need an update: https://snyk.io/docs/faqs search proxy.

This has been manually tested:

DEBUG=snyk:req ./node_modules/.bin/snyk protect  --debug
```
...
Successfully applied Snyk patches
...
snyk:req Using proxy:  http://proxyout.companydomain.com:8080 +0ms
```

#### Where should the reviewer start?
?

#### How should this be manually tested?
block internet access, except through a proxy and set the proxy as one of the needed environment variables.

#### Any background context you want to provide?
I had a support chat with Aviat and Ellen and here's the error I gave them:

```
$ ./node_modules/.bin/snyk protect  --debug
  snyk protect { _: [ [Circular] ], debug: true } +0ms
  snyk ~~~~ LIVE RUN ~~~~ +266ms
  snyk checking for cli updates +4ms
{ url: 'https://snyk.io/api/v1/vuln/npm/patches',
  json: true,
  headers: { 'x-is-ci': false, authorization: 'token undefined' } }
  snyk { url: 'https://snyk.io/api/v1/vuln/npm/patches',
  snyk   json: true,
  snyk   headers: { 'x-is-ci': false, authorization: 'token undefined' } } +65ms
  snyk add local true +1ms
  snyk policies found [ '/home/jenkins/slave/workspace/cmc-citizen-frontend_PR-171-AWDTYLFSQAZX3ZN5LUZLPKFADT667EZMLGNTHHXTHDKW6IQORVBQ',
  '/home/jenkins/slave/workspace/cmc-citizen-frontend_PR-171-AWDTYLFSQAZX3ZN5LUZLPKFADT667EZMLGNTHHXTHDKW6IQORVBQ/node_modules/snyk' ] +2s
  snyk add policies 2 +0ms
  snyk add packageManager npm +4ms
  snyk add packageName cmc-citizen-frontend +0ms
  snyk add packageVersion 1.1.0 +0ms
  snyk add package cmc-citizen-frontend@1.1.0 +0ms
  snyk add success false +150msbase...
▽
  snyk add error-message connect ECONNREFUSED 23.37.79.107:443 +0ms
 
▽
  snyk add error Error: connect ECONNREFUSED 23.37.79.107:443
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1182:14) +0ms
  snyk add error-code ECONNREFUSED +0ms
  snyk add command protect +0ms
Error: connect ECONNREFUSED 23.37.79.107:443
    at Object._errnoException (util.js:1024:11)
    at _exceptionWithHostPort (util.js:1046:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1182:14)
  snyk { command: 'bad-command',
  snyk   args: [ { debug: true } ],
  snyk   metadata:
  snyk    { local: true,
  snyk      policies: 2,
  snyk      packageManager: 'npm',
  snyk      packageName: 'cmc-citizen-frontend',
  snyk      packageVersion: '1.1.0',
  snyk      package: 'cmc-citizen-frontend@1.1.0',
  snyk      success: false,
  snyk      'error-message': 'connect ECONNREFUSED 23.37.79.107:443',
  snyk      error: 'Error: connect ECONNREFUSED 23.37.79.107:443\n    at Object._errnoException (util.js:1024:11)\n    at _exceptionWithHostPort (util.js:1046:20)\n    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1182:14)',
  snyk      'error-code': 'ECONNREFUSED',
  snyk      command: 'protect' },
  snyk   version: '1.49.2',
  snyk   os: 'Linux 3.10',
  snyk   nodeVersion: 'v8.9.1',
  snyk   id: 'f28b66bd647654aa4de26daf12e60d28f8cdc677',
  snyk   ci: false } +2ms
  snyk Error: connect ECONNREFUSED 23.37.79.107:443
  snyk     at Object._errnoException (util.js:1024:11)
  snyk     at _exceptionWithHostPort (util.js:1046:20)
  snyk     at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1182:14) +8ms
```

#### What are the relevant tickets?
None, feel free to create one if needed

### CODE REVIEW

File: lib/request/request.js [hunk 1]
```diff
@@ -12,6 +12,13 @@ var querystring = require('querystring');
 var zlib = require('zlib');
 var config = require('../config');
 
+function resolveProxy() {
+  return process.env.https_proxy
+      || process.env.HTTPS_PROXY
+      || process.env.http_proxy
+      || process.env.HTTPS_PROXY;
```
Comment (geek-soft) [inline lib/request/request.js]: A `process.env.HTTPS_PROXY` it used twice. I think you meant `process.env.HTTP_PROXY`

Comment (timja-kainos) [reply lib/request/request.js]: Fixed the typo

**Comment by timja-kainos** [comment]
All the builds appear to be failing because they require an authenticated account,
```
snyk test requires an authenticated account. Please run `snyk auth` and try again.
```

I assume it's because I forked the repo and don't have trust?

**Comment by timja-kainos** [comment]
@aviadatsnyk nice find on the library, I did have a quick look around no_proxy but didn't see anything.

I've made the changes and tested manually that no_proxy works,
used:
```
export no_proxy="snyk.io"
DEBUG=snyk:req ./node_modules/.bin/snyk protect  --debug

$ env | grep proxy
HTTPS_PROXY=http://proxyout.companydomain.com:8080
no_proxy=snyk.io
```

**Pull request closed without merging**